src/extension/vectortiles/src/main/java/org/geotools/renderer/lite/VectorMapRenderUtils.java

```diff
@@ -105,10 +110,99 @@ public static Query getStyleQuery(Layer layer, WMSMapContent mapContent) throws
         query.setProperties(Query.ALL_PROPERTIES);
 
         Hints hints = query.getHints();
+
         hints.put(Hints.FEATURE_2D, Boolean.TRUE);
 
+        // update the screenmaps
+        try {
+            CoordinateReferenceSystem crs = schema.getCoordinateReferenceSystem();
+            if (crs != null) {
+                FeatureSource<FeatureType, Feature> source =
+                        (FeatureSource<FeatureType, Feature>) layer.getFeatureSource();
+                Set<RenderingHints.Key> fsHints = source.getSupportedHints();
+                CoordinateReferenceSystem destSRS = query.getCoordinateSystemReproject();
+                MathTransform sourceToScreen =
+                        buildFullTransform(
+                                crs,
+                                destSRS != null ? destSRS : crs,
+                                mapContent.getRenderingTransform());
+                double[] spans =
+                        // fallback, use the entire rendering area
+                        Decimator.computeGeneralizationDistances(
+                                sourceToScreen.inverse(), screenSize, generalizationDistance);
+                for (LiteFeatureTypeStyle fts : styleList) {
+                    if (fts.screenMap != null) {
+                        fts.screenMap.setTransform(sourceToScreen);
+                        fts.screenMap.setSpans(spans[0], spans[1]);
+                        if (fsHints.contains(Hints.SCREENMAP)) {
+                            // replace the renderer screenmap with the hint, and avoid doing
```

Comment:
Code should be updated to match the vector tiles context (there is no "renderer screenmap", but the code could be referring to the PreProcess pipeline step screenmap instead).